### PR TITLE
Doxygen code examples for the HTTP client API.

### PR DESCRIPTION
**Pull request by sarenameas**
This PR brings in code examples for each of the HTTP client API.

image

### CODE REVIEW

File: source/include/core_http_client.h [hunk 1]
```diff
@@ -663,7 +735,7 @@ HTTPStatus_t HTTPClient_AddRangeHeader( HTTPRequestHeaders_t * pRequestHeaders,
  * - #HTTP_NO_RESPONSE (No data was received from the transport interface.)
  * - #HTTP_INSUFFICIENT_MEMORY (The response received could not fit into the response buffer
  * or extra headers could not be sent in the request.)
- * - #HTTP_PARSER_INTERNAL_ERROR (Internal parsing error.)
+ * - #HTTP_PARSER_INTERNAL_ERROR (Internal parsing error.)<br><br>
```
Comment (yourslab) [inline source/include/core_http_client.h:738]: Could these be replaced with actual newlines? These tags just look a little out of place.

File: source/include/core_http_client.h [hunk 2]
```diff
@@ -672,6 +744,46 @@ HTTPStatus_t HTTPClient_AddRangeHeader( HTTPRequestHeaders_t * pRequestHeaders,
  * - #HTTP_SECURITY_ALERT_INVALID_STATUS_CODE
  * - #HTTP_SECURITY_ALERT_INVALID_CHARACTER
  * - #HTTP_SECURITY_ALERT_INVALID_CONTENT_LENGTH
+ *
+ * <b>Example</b>
```
Comment (yourslab) [inline source/include/core_http_client.h:748]: Would the markdown equivalent work here too? \*\*Example\*\*

File: source/include/core_http_client.h [hunk 3]
```diff
@@ -672,6 +744,46 @@ HTTPStatus_t HTTPClient_AddRangeHeader( HTTPRequestHeaders_t * pRequestHeaders,
...
+ * @code{c}
+ * // Variables used in this example.
+ * HTTPStatus_t httpLibStatus = HTTP_SUCCESS;
+ * TransportInterface_t transportInterface = { 0 };
+ * HTTPResponse_t = { 0 };
+ * char requestBody[] = "This is example request body.";
```
Comment (yourslab) [inline source/include/core_http_client.h:754]:
```suggestion
 * char requestBody[] = "This is an example of a request body.";
```

File: source/include/core_http_client.h [hunk 4]
```diff
@@ -672,6 +744,46 @@ HTTPStatus_t HTTPClient_AddRangeHeader( HTTPRequestHeaders_t * pRequestHeaders,
...
+ *
+ * // Assume that requestHeaders has been initialized with
+ * // HTTPClient_InitializeResponseHeaders() and any headers needed have been
```
Comment (yourslab) [inline source/include/core_http_client.h:757]:
```suggestion
 * // HTTPClient_InitializeResponseHeaders() and any desired headers have been
```

Needed seems to indicate that the call to `HTTPClient_AddHeader` is required for the request.

File: source/include/core_http_client.h [hunk 5]
```diff
@@ -705,6 +817,24 @@ HTTPStatus_t HTTPClient_Send( const TransportInterface_t * pTransport,
  * - #HTTP_HEADER_NOT_FOUND (Header is not found in the passed response buffer.)
  * - #HTTP_INVALID_RESPONSE (Provided response is not a valid HTTP response for parsing.)
  * - #HTTP_PARSER_INTERNAL_ERROR(If an error in the response parser.)
+ *
+ * <b>Example</b>
+ * @code{c}
+ * HTTPStatus_t httpLibStatus = HTTP_SUCCESS;
```
Comment (yourslab) [inline source/include/core_http_client.h:823]: Our style guide states: `Names should avoid using abbreviations.`

However, this is sample code and probably does not need to reflect what the style guide states (We aren't using multi-line comments either). I leave it up to you.

File: source/include/core_http_client.h (hunk 1 above)
Comment (sarenameas) [reply source/include/core_http_client.h:738]: sure

File: source/include/core_http_client.h (hunk 5 above)
Comment (sarenameas) [reply source/include/core_http_client.h:823]: Good luck policing "cert"

Comment (yourslab) [reply source/include/core_http_client.h:823]: I have no idea why we make up rules that are meant to be broken

File: source/include/core_http_client.h (hunk 4 above)
Comment (sarenameas) [reply source/include/core_http_client.h:757]: Sure ill write "additional"

File: source/include/core_http_client.h (hunk 2 above)
Comment (sarenameas) [reply source/include/core_http_client.h:748]: sure

**Review by yourslab — APPROVED** [review]

**Review by n9wxu — APPROVED** [review]

**Pull request merged**